python/ray/experimental/workflow/storage/base.py

```diff
@@ -39,6 +80,33 @@ class Storage(metaclass=abc.ABCMeta):
     """
 
     @abstractmethod
+    def _make_key(self, *names: str) -> str:
```

Comment:
As discussed, this `Storage` involves too many app-level logics. We can move this to other places.